lib/ddtrace/contrib/redis/patcher.rb

```diff
@@ -2,9 +2,22 @@
 
 module Datadog
   module Contrib
+    # Redis contains the patcher and a method for determining the service name
     module Redis
-      SERVICE = 'redis'.freeze
-      DRIVER = 'redis.driver'.freeze
+      DEFAULT_SERVICE_NAME = 'redis'.freeze
+
+      def self.service_name
+        @service_name ||= \
+          if defined?(::Sinatra)
```

Comment:
Rather than a bunch of `defined?(...)` blocks, how about passing the initial config to the `patch` method below?

I opened an issue for refactoring config injection into the instrumenters here: https://github.com/DataDog/dd-trace-rb/issues/188. That ties into how you could then configure this.

Reply:
that sounds good, I'd much prefer it to the if defined? treatment.

Reply:
Yes we're planning to improve our way to configure integrations like @nerdrew did for Redis. We're evaluating the implications of that big change in our `Monkey` module, and then we'll be back with a plan.